tv_apps/smart-system/js/app_install_manager.js

```diff
@@ -292,6 +322,72 @@ var AppInstallManager = {
     app.onprogress = this.handleProgress;
   },
 
+  previewApp: function ai_previewApp(app) {
+    if (!this.isMarketPlaceAppActive()) {
+      return;
+    }
+
+    var marketplace = AppWindowManager.getActiveApp();
+    var manifestURL = app.manifestURL;
+    var appURL = app.origin +
+      (app.manifest.launch_path === '/' ? '/' : app.manifest.launch_path);
+
+    marketplace.childWindowFactory.createPreviewWindow({
+      detail: {
+        url: appURL,
+        origin: app.origin,
+        manifestURL: manifestURL
+      }
+    });
+
+    marketplace.element.addEventListener('_closed', (evt) => {
+      this.clearPreviewApps();
+    });
+
+    var previewAppWindow = marketplace.getTopMostWindow();
+    var previewContent = previewAppWindow.element;
+    var msg = {
+      title: 'Press OPTION to',
+      text: 'Add to Apps'
+    };
+
+    previewContent.addEventListener('_opened', (evt) => {
+      var TYPE = window.InteractiveNotifications.TYPE;
+      window.interactiveNotifications.showNotification(TYPE.TOAST, msg);
+    });
+  },
+
+  getPreviewApps: function() {
```

Comment:
I personally don't like the name `getPreviewApps` since the term `app` usually indicates an app instance in system app, not just a manifestURL. How about something like `getPreviewAppManifestURL`?